### PR TITLE
sql_query_for_missing_files was not working on just missing files, bu…

### PR DESCRIPTION
**Pull request by azurebrd**
…t workflow tag status, renamed function.  subquery LEFT JOIN referencefile was making it very slow, instead processing the reference list subset to display, and query just that set for referencefile information, then aggregate that into the resulting ref_data to return.

Also, the default query searches for either of two specific ATP terms, instead of the exclusion of two other specific ATP terms

### CODE REVIEW

File: agr_literature_service/api/crud/reference_crud.py [hunk 1]
```diff
@@ -864,7 +854,36 @@ def sql_query_for_missing_files(db: Session, mod_abbreviation: str, order_by: st
...
         params['offset'] = str(offset)
 
-    return query.bindparams(**params)
+    rows = db.execute(query, params).mappings().fetchall()
+    ref_data = jsonable_encoder(rows)
+    if not ref_data:
+        return ref_data
+
+    reference_ids = [item['reference_id'] for item in ref_data if 'reference_id' in item]
+    reference_ids_sql_txt = ', '.join(str(item) for item in reference_ids)
```
Comment (sweng66) [inline agr_literature_service/api/crud/reference_crud.py:863]: Can we  eliminate the intermediate reference_ids_sql_txt and directly use bind parameter as follows? 

reference_ids = [item['reference_id'] for item in ref_data if 'reference_id' in item]
reffile_query_str = """
    SELECT reference_id,
           COUNT(1) FILTER (WHERE file_class = 'main') AS maincount,
           COUNT(1) FILTER (WHERE file_class = 'supplement') AS supcount
    FROM referencefile
    WHERE reference_id = ANY(:reference_ids)
    GROUP BY reference_id
"""

rows_reffile = db.execute(text(reffile_query_str), {'reference_ids': reference_ids}).mappings().fetchall()

**Review by sweng66 — APPROVED** [review]
Looks great! Thanks @azurebrd

File: agr_literature_service/api/crud/reference_crud.py (hunk 1 above)
Comment (azurebrd) [reply agr_literature_service/api/crud/reference_crud.py:863]: Oh, cool !  I didn't know the syntax for that, that looks much better, I'll try it after stand up, thanks !

Comment (sweng66) [reply agr_literature_service/api/crud/reference_crud.py]: The fix looks great. Thanks

**Review by sweng66 — APPROVED** [review]

**Comment by azurebrd** [comment]
Thanks @sweng66 !

**Pull request merged**